scenario_generator/config/templates/sim_launch_template.sh.j2

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+#  Copyright (C) 2025 LEIDOS.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+#  use this file except in compliance with the License. You may obtain a copy of
+#  the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#  License for the specific language governing permissions and limitations under
+#  the License.
+
+# Directory paths
+CDASIM_CONFIG_DIR={{ cdasim_config_dir }}
+STREET_CONFIG_DIR={{ street_config_dir }}
+CARMA_CONFIG_DIR={{ carma_config_dir }}
+
+# Store current directory
+CURRENT_DIR=$(pwd)
+
+# Track initialized projects for cleanup
+PROJECT_NAMES=()
+
+# Cleanup function to shut down initialized Docker Compose projects
+cleanup() {
+    echo "Cleaning up initialized Docker Compose projects..."
+    for project in "${PROJECT_NAMES[@]}"; do
+        if [ -n "$project" ]; then
+            echo "Shutting down project: $project"
+            docker compose -p "$project" down || true
+        fi
+    done
+    # Remove sim_net network
+    docker network rm sim_net || true
+    # Return to original directory
+    cd "$CURRENT_DIR" || exit 1
+    exit 1
+}
+
+# Trap errors to trigger cleanup
+trap cleanup ERR
+
+# Create shared sim_net
+docker network create sim_net || true
+
+# Launch CDASim
+cd "$CDASIM_CONFIG_DIR" || { echo "Failed to cd to $CDASIM_CONFIG_DIR"; cleanup; }
+(source .env.cdasim && docker compose -f docker-compose.yml --env-file .env.cdasim -p "{{ config['env_settings']['cdasim']['PROJECT_NAME'] }}" up -d && PROJECT_NAMES+=("{{ config['env_settings']['cdasim']['PROJECT_NAME'] }}"))
+
+# Launch vehicles (carma_config includes platform)
+{% for vehicle in vehicles %}
+cd "$CARMA_CONFIG_DIR" || { echo "Failed to cd to $CARMA_CONFIG_DIR"; cleanup; }
+(source .env.vehicle_{{ loop.index }} && docker compose -f docker-compose.yml --env-file .env.vehicle_{{ loop.index }} -p "{{ vehicle.PROJECT_NAME }}" up -d && PROJECT_NAMES+=("{{ vehicle.PROJECT_NAME }}"))
+# Create platform network for this vehicle
+docker network create {{ vehicle.PROJECT_NAME }}_platform_net || true
+{% endfor %}
+
+# Launch CARMA Streets
+{% for street in streets %}
+cd "$STREET_CONFIG_DIR" || { echo "Failed to cd to $STREET_CONFIG_DIR"; cleanup; }
+(source .env.street_{{ loop.index }} && docker compose -f docker-compose.yml --env-file .env.street_{{ loop.index }} -p "{{ street.PROJECT_NAME }}" up -d && PROJECT_NAMES+=("{{ street.PROJECT_NAME }}"))
+# Create street network for this instance
+docker network create {{ street.PROJECT_NAME }}_street_net || true
+{% endfor %}
+
+# Return to original directory
+cd "$CURRENT_DIR" || exit 1
```

Comment:
Would be good to have a final echo here to indicate that the launch has completed successfully